### DISCUSSION TITLE
Reporting some bugs

### DISCUSSION THREAD

**Discussion by intellq** [RainbowMiner/RainbowMiner#2163] [Q&A] [answered]
1) RVN balance at 2miners is bugged. For some reason, the "2MinersAE BTC" balance is the original balance divided by 10:
![rvn](https://user-images.githubusercontent.com/22642001/191252853-d9c694b1-45c2-4fdc-8264-ab83bf307650.jpg)

2) Even after correcting this balance bug, the total balance is not calculating as it's supposed to. The "total pools" balance would be doubled, because rainbow is thinking I have balance in 2Miners and 2MinersAE (I dont even know what this is) and adding both to the total. This problem is prolly cause I'm using a BTC wallet for all 2Miners RVN/ETC/ERG.

3) Since 4.8.6.0 the keyboard input is bugged when the mining proccess starts. After some time, any keyboard command doesn't work, It's not working x (to exit), v (for verbose), etc. Even ctrl+c doesn't work. Maybe it's a powershell thing, cause this only happens with 7.2.6. Using 5.0 and the problem is gone (tested in 3 different machines).

**Comment by intellq** [comment]
About the item 3, I think I discovered what is going on. In my case, I'm using Teamred right now, and if I press "Q" in rainbowminer (during mining proccess), there's a message for Teamred crashing. But it's not crashing, for some reason it's like the keyboard focus is stealed from powershell and keyboard commands are being given for the miner -- even there's no window present for Teamred (ShowMinerWindow = "0").

And this is happening because the "Q" key tells Teamred to quit. All my keyboard commands are going to the miner hidden window, not to powershell window.

btw, I forgot to tell, the problem is only in Windows

**Reply by RainbowMiner** [reply]
Thank you for the details - so Teamred is kind of greedy with the keyboard inputs. I'll try to figure something out, so that we can work around this "feature".

**Comment by RainbowMiner** [comment, marked as answer]
Regarding 1. and 2. 
- I think you are using your BTC wallet address as "ETC", "RVN", "ERG" entries in "2Miners" in pools.config.txt. This is the wrong approach, "2Miners" and "2MinersSolo" are thought as pool modules for direct coin mining. E.g. mine ERG -> get ERG, mine RVN -> get RVN ...
- to make use of 2Miner's auto-exchange feature (mine ETC -> get BTC, mine RVN -> get BTC, mine ERG -> get BTC) please use our "2MinersAE" pool module. Just input your BTC wallet address in "BTC" (2MinersAE section in pools.config.txt), that's it. You can even select, which coins to mine in parameter "CoinSymbol".

But - and here comes the bug, 2MinersAE shouldn't show up with the balances, if you haven't selected it in "PoolName". I'll fix this.

**Comment by intellq** [comment]
I didn't know about 2MinersAE being the "auto exchange" wallet. Changed my configs, thanks.

But one thing is still bugged: RVN balance is incorrect. It's displaying an incorrect value: the real value divided by 10, like I showed in the picture in original post.

**Reply by RainbowMiner** [reply]
Yes, I see. I'll fix this asap.

**Reply by RainbowMiner** [reply]
Ok, fixed with this commit: e84bd971db6774f528f33ceb8d32029590554b22

**Comment by solution234** [comment]
I have tried to recreate the bug here: https://rebrand.ly/lub4f9a
